src/_pytest/compat.py

```diff
@@ -324,4 +324,7 @@ class FuncargnamesCompatAttr:
     @property
     def funcargnames(self):
         """ alias attribute for ``fixturenames`` for pre-2.3 compatibility"""
-        return self.fixturenames
```

Comment:
We dropped the ball on this one, we should have deprecated it properly sooner. 

Unfortunately, according to our deprecation policy, we should always deprecate something for at least two minor releases before removing it on a major release. Even if it is an old attribute I'm sure there old code out there still using it, so we should follow the normal deprecation path. 

Having said that, I believe the right path would be to change this to issue a `pytest.PytestDeprecationWarning` instead. 😕

Reply:
Fair enough... converted to a deprecation.